### PR TITLE
Make sure game / players are dirtied when a game connection ends

### PR DESCRIPTION
**Pull request by Crotalus**
Make sure the cleanup of a game connection gets done regardless of eventual write exceptions to a peer.

### CODE REVIEW

**Comment by micheljung** [comment]
Fixes #200

**Comment by Crotalus** [comment]
Hopefully yes, only a theory from my side though while parsing the code. If someone with access to logs could check if the server sometime spits out exception in send_DisconnectToPeer (I guess lobbyconnection.send() at the lower level) that would be nice.

File: server/gameconnection.py [hunk 1]
```diff
@@ -433,7 +433,10 @@ def abort(self, logspam=''):
             if self.game.state == GameState.LOBBY:
                 for peer in self.game.connections:
                     if peer != self:
-                        peer.send_DisconnectFromPeer(self.player.id)
+                        try:
+                            peer.send_DisconnectFromPeer(self.player.id)
+                        except:
```
Comment (Sheeo) [inline server/gameconnection.py]: We definitely don't want to just pass this and not at least log it :)

File: server/gameconnection.py [hunk 2]
```diff
@@ -433,7 +433,10 @@ def abort(self, logspam=''):
             if self.game.state == GameState.LOBBY:
                 for peer in self.game.connections:
                     if peer != self:
-                        peer.send_DisconnectFromPeer(self.player.id)
+                        try:
+                            peer.send_DisconnectFromPeer(self.player.id)
+                        except Exception as ex: # pragma no cover
+                            self.log.exception("peer_sendDisconnectFromPeer failed for player %i", self.player.id)
```
Comment (Sheeo) [inline server/gameconnection.py]: Can we log the actual exception value, too?

Comment (Crotalus) [inline server/gameconnection.py]: I thought self.log.exception did this automatically? I may be wrong though

Comment (Crotalus) [inline server/gameconnection.py]: I guess it just calls  sys.exc_info() internally to get the exception?

Comment (micheljung) [inline server/gameconnection.py]: hm ok, maybe you're right then x)

**Pull request closed without merging**